packages/eslint-plugin/tests/rules/no-misused-promises.test.ts

```diff
@@ -500,6 +500,462 @@ foo(bar);
       },
       options: [{ checksVoidReturn: { attributes: true } }],
     },
+    // #region checksVoidReturn.subtypes
```

Comment:
Hope the amount of test cases here wasn't too overkill. I actually did find them useful for catching random cases (e.g. type intersections, conditional types, class expressions) that weren't properly handled by my original implementation(s).

Reply:
IME as long as the tests are hand-written and not auto-generated, they're almost always not overkill. Nice coverage!